src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
+            validators = validators.Concat(AddOptionIfAvailable(command, BaseUriOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, LocationOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, JobNameOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, ShotsOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, OutputOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, DryRunOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, VerboseOption));
+            validators = validators.Concat(MarkOptionsAsMutuallyExclusive(
+                command,
+                new[] { BaseUriOption.Aliases.First(), LocationOption.Aliases.First() }));
+
+            return (command, validators);
+        }
 
         /// <summary>
-        /// The output option.
+        /// Creates the simulation option for the simulate command with the given entry point.
         /// </summary>
-        internal static readonly OptionInfo<OutputFormat> OutputOption = new OptionInfo<OutputFormat>(
-            ImmutableList.Create("--output"),
-            OutputFormat.FriendlyUri,
-            "The information to show in the output after the job is submitted.");
+        /// <param name="entryPoint">The entry point.</param>
+        /// <returns>The OptionInfor for the simulator option.</returns>
+        private OptionInfo<string> CreateSimulatorOption(IEntryPoint entryPoint) =>
+            new OptionInfo<string>(
+                this.settings.SimulatorOptionAliases,
+                entryPoint.DefaultSimulatorName,
+                "The name of the simulator to use.",
+                suggestions: new[]
+                {
+                    this.settings.QuantumSimulatorName,
+                    this.settings.ToffoliSimulatorName,
+                    this.settings.ResourcesEstimatorName,
+                    entryPoint.DefaultSimulatorName
+                });
 
         /// <summary>
-        /// The dry run option.
+        /// Creates the target option for the Azure submit command with the given entry point.
         /// </summary>
-        internal static readonly OptionInfo<bool> DryRunOption = new OptionInfo<bool>(
-            ImmutableList.Create("--dry-run"),
-            false,
-            "Validate the program and options, but do not submit to Azure Quantum.");
+        /// <param name="entryPoint">The entry point.</param>
+        /// <returns>The OptionInfo for the --target option.</returns>
+        private OptionInfo<string?> CreateTargetOption(IEntryPoint entryPoint)
+        {
+            var targetAliases = ImmutableList.Create("--target");
+            const string targetDescription = "The target device ID.";
+            return string.IsNullOrWhiteSpace(entryPoint.DefaultExecutionTarget)
```

Comment:
Same as above: the default execution target is a project setting instead of an entry-point-specific setting.

Reply:
done